src/me.rs

```diff
@@ -397,15 +452,15 @@ pub struct FullSearch {}
 impl MotionEstimation for DiamondSearch {
   fn full_pixel_me<T: Pixel>(
     fi: &FrameInvariants<T>, fs: &FrameState<T>, rec: &Arc<ReferenceFrame<T>>,
-    po: &PlaneOffset, bo: &BlockOffset, lambda: u32, ref_slot: usize,
+    po: &PlaneOffset, bo: &BlockOffset, lambda: u32,
     cmv: MotionVector, pmv: [MotionVector; 2], mvx_min: isize, mvx_max: isize,
     mvy_min: isize, mvy_max: isize, blk_w: usize, blk_h: usize,
     best_mv: &mut MotionVector, lowest_cost: &mut u64, ref_frame: usize
   ) {
-    let frame_mvs = &fs.frame_mvs[ref_slot];
+    let frame_mvs = &fs.frame_mvs[ref_frame - LAST_FRAME];
```

Comment:
I know this pattern is from libaom.... but I hate it :)

I'll change the whole codebase in my own patch.